src/Oscoin/Consensus/Nakamoto.hs

```diff
@@ -200,6 +202,16 @@ chainDifficulty (NonEmpty.fromList -> blks)
     | otherwise =
         pure prevDifficulty
   where
+
+    -- The difficulty is adjusted every 'difficultyBlocks' blocks, and only if we do have
+    -- 2016 blocks at hand to compute the new one. As the 'height' of genesis
+    -- is 0, we need to add 1 to the value returned by 'blockHeight', to be
+    -- sure to trigger an adjustment every 'difficultyBlocks' blocks.
+    adjustmentRequired =
+        (bHeight (NonEmpty.head blks) + 1 ) `mod` fromIntegral difficultyBlocks == 0
+        && NonEmpty.length blks == fromIntegral difficultyBlocks
```

Comment:
Looks good, I think this is correct. Another way to formulate this would be "block height > 0 && block height % 2016 == 0`

Reply:
Yeah, possibly -- in my formulation I wanted to stress the fact that non only we need to trigger this every 2016 blocks, but that we likely want also _exactly_ (at least?) `difficultyBlocks` blocks as input, in order to make an informative decision (otherwise we cannot compute a proper `rangeEnd` and `rangeStart`).